resources/views/livewire/memory-notes/memory-notes-list-item.blade.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+use App\Platform\Livewire\Forms\MemoryNoteForm;
+use Illuminate\Support\Facades\Auth;
+
+use function Livewire\Volt\{on, state};
+
+// == props
+state(['memoryNote']); // MemoryNote
+
+// == state
+state(['isEditing' => false]);
+state(['wasDeleted' => false]);
+
+// == actions
+
+// == lifecycle
+on(['editComplete-{memoryNote.id}' => function() {
+    $this->isEditing = false;
+}]);
+
+on(['deleteComplete-{memoryNote.id}' => function() {
+    $this->wasDeleted = true;
+}]);
+
+?>
+
+<tr
+    class="[&>td]:align-top"
+    :class="{ 'hidden': wasDeleted }"
+    x-data="{
+        isEditing: $wire.entangle('isEditing'),
+        wasDeleted: $wire.entangle('wasDeleted'),
+    }"
+>
+    <td>
+        <p class="font-mono font-bold sticky top-11">
+            {{ $memoryNote->address_hex }}
+        </p>
+    </td>
+    
+    <td width="100%">
+        <div x-show="isEditing" x-cloak>
+            <livewire:memory-notes.memory-note-edit-form
+                :memoryNote="$memoryNote"
+            />
+        </div>
+
+        <p class="font-mono" x-show="!isEditing">
+            {!! nl2br(e($memoryNote->body)) !!}
+        </div>
+    </td>
+    
+    <td>
+        <div class="flex justify-center py-1">
+            {!! 
+                userAvatar(
+                    $memoryNote->user->display_name,
+                    label: false,
+                    iconSize: 24,
+                    iconClass: 'rounded-sm'
+                )
+            !!}
+        </div>
+    </td>
+    
+    @can('manage', App\Models\MemoryNote::class)
+        <td class="text-right min-w-[80px]">
+            @can('update', $memoryNote)
+                <div class="mt-1">
+                    <button
+                        class="btn"
+                        x-show="isEditing"
+                        x-on:click="isEditing = false"
+                        x-cloak
+                    >
+                        Cancel
+                    </button>
+                    
+                    <button
+                        class="btn"
+                        x-show="!isEditing"
+                        x-on:click="isEditing = true"
+                    >
+                        Edit
+                    </button>
```

Comment:
Not sure where it's coming from, or if it even matters, but every row is generating an invisible Save button:
```
            <button
                class="btn"
                x-on:click="window.beginSave(window.Livewire.find('YeBoRjAofI7XmqZ8YrOb'), )"
                disabled
                wire:dirty.remove.attr="disabled"
            >
                Save
            </button>
```

Reply:
It could definitely be more efficient. The offending code was here:

```html
<div x-show="isEditing" x-cloak>
    <livewire:memory-notes.memory-note-edit-form
        :memoryNote="$memoryNote"
    />
</div>
```

In a Livewire component, we can conditionally render stuff in real-time with our standard Blade directives such as `@if ()`. However, this is quite slow. Since this is driven by a client-side action, it's better to handle this with Alpine.js.

`x-show` is an Alpine directive that applies an inline style of "display: none;" unless its given condition is met. What this means though is that its DOM nodes are always present.

Here's my updated implementation:
```html
<div x-show="isEditing">
    <template x-if="isEditInitialized">
        <livewire:memory-notes.memory-note-edit-form
            :memoryNote="$memoryNote"
        />
    </template>
</div>
```

Now we're combining `x-show` and `x-if`. `x-if` is very similar to `x-show`, but it directly adds and removes elements from the DOM. The DOM elements aren't present, they're stored efficiently in memory as a `DocumentFragment` (and are still going to be visible in most browser devtools, but will likely be labeled differently).

The nuance here is we can't rely on `x-if` alone to remove those DOM nodes when the user exits Edit Mode. Unfortunately, this breaks Livewire's data binding. With the combination of `x-if` and `x-show`, after the first render in Edit Mode, we return to pure client-side visibility control with CSS.

Hopefully this explanation makes sense.

Reply:
Something in this change is causes the submissions (delete and update) to not refresh the page/row.

The first note _has been_ deleted. The second _has been_ saved.
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/8c99fc69-aa9d-4274-ac1e-5158b867d8bb)

Reply:
For several hours today, I attempted to mitigate this issue while retaining the `x-if` directive, but I haven't had any luck.

I've reverted the `x-if` change. Having those hidden buttons probably isn't a big deal for the time being.

One other observation I've made: some routes such as `/game/2721/notes` crash my page locally. This happens on `master` without this PR's changes as well. At some point, for huge lists of notes, we may want to implement pagination & search.